homeassistant/helpers/intent.py

```diff
@@ -438,6 +439,30 @@ def _filter_by_name(
                     break
 
 
+def _matches_name(
+    state: State,
+    name: str,
+    entities: entity_registry.EntityRegistry,
+) -> bool:
+    """Return true if entity's (normalized) name or alias matches the provided name."""
+    name_norm = _normalize_name(name)
+
+    if (state.entity_id == name) or _normalize_name(state.name) == name_norm:
```

Comment:
Why are we comparing it to the `entity_id` in the first place? Users should call their entities by their name, not id.

Reply:
This allows intent recognizers to target specific entities when there would otherwise be ambiguity. For example, I map "TV" to a specific entity id in a pipeline.

Reply:
Does that actually happen anywhere though? It would have to happen already at the sentences level and I don't think we have any sentences with a name slot hard-coded with an `entity_id`.

Unless I'm misunderstanding what you mean by:
>For example, I map "TV" to a specific entity id in a pipeline.

Reply:
The default intents no longer to this, you're correct (they did originally). I want to leave it in for now, though, as other intent recognizers may use it. I'm not concerned about overlap, as entity ids are very distinct from human names.
A potential problem is with area/floor ids, though. If you have an area named "kitchen" but it's area id is "bedroom" (for some reason), you could match the wrong one.